src/main/java/com/cloudbees/jenkins/plugins/advisor/casc/AdvisorRootConfigurator.java

```diff
@@ -0,0 +1,151 @@
+            if (excludedCN instanceof Sequence) {
+                Sequence s = (Sequence) excludedCN;
+                for (CNode cNode : s) {
+                    excludedComponents.add(cNode.asScalar().getValue());
+                }
+
+                if (excludedComponents.isEmpty()) {
+                    excludedComponents.add(AdvisorGlobalConfiguration.SEND_ALL_COMPONENTS);
+                }
+            } else {
+                throw new ConfiguratorException(this, "Excluded components are expected to be a collection.");
+            }
+
+        } else {
+            excludedComponents.add(AdvisorGlobalConfiguration.SEND_ALL_COMPONENTS);
+        }
+
+        // Check and apply configuration
+        AdvisorGlobalConfiguration insights = getTargetComponent(configurationContext);
+        AdvisorGlobalConfiguration.DescriptorImpl descriptor = (AdvisorGlobalConfiguration.DescriptorImpl) insights.getDescriptor();
+        boolean valid = true;
+        if (descriptor.doCheckEmail(email).kind.equals(FormValidation.Kind.ERROR) || descriptor.doCheckCc(cc).kind.equals(FormValidation.Kind.ERROR)) {
+            valid = false;
+        }
+        if (valid) {
+            if (acceptToS) {
+                updateConfiguration(insights, email, cc, true, nagDisabled, excludedComponents);
+            } else {
+                // In UI, if you don't accept the ToS, the configuration is not applied, so here we throw an exception
+                throw new ConfiguratorException(this, "Terms of Service for CloudBees Jenkins Advisor have to be accepted. Please, review the yaml file.");
+            }
+        } else {
+            // In UI, if the fields are invalid, the configuration is not applied, so here we throw an exception
+            throw new ConfiguratorException(this, "Invalid configuration for CloudBees Jenkins Advisor. Please, review the content of email and cc fields in the yaml file.");
+        }
+        return insights;
+    }
+
+    private void updateConfiguration(AdvisorGlobalConfiguration conf, String email, String cc, boolean acceptToS, boolean nagDisabled, Set<String> excludedComponents) {
+        conf.setEmail(email);
+        conf.setCc(cc);
+        conf.setAcceptToS(acceptToS);
+        conf.setNagDisabled(nagDisabled);
+        conf.setExcludedComponents(excludedComponents);
+        conf.setValid(true);
+    }
+
+    @Override
+    public AdvisorGlobalConfiguration getTargetComponent(ConfigurationContext configurationContext) {
+        return getConfiguration();
+    }
+
+    @Override
+    public Class<AdvisorGlobalConfiguration> getTarget() {
+        return AdvisorGlobalConfiguration.class;
+    }
+
+    @CheckForNull
+    @Override
+    public CNode describe(AdvisorGlobalConfiguration instance, ConfigurationContext context) throws Exception {
+        Mapping mapping = new Mapping();
+        // In UI, email is mandatory. If the email is empty, the Advisor is not configured, so nothing should be exported
```

Comment:
Does an empty mapping means that no section is generated inside the exported yaml?

Reply:
yes. if the email is not configured, then the exported yaml will not have this section.